test/physics/em/RelativisticBrem.test.cc

```diff
@@ -0,0 +1,353 @@
+
+    RandomEngine& rng_engine = this->rng();
+
+    // Produce four samples from the original incident angle/energy
+    std::vector<double> angle;
+    std::vector<double> energy;
+
+    for (int i : celeritas::range(num_samples))
+    {
+        Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+
+        EXPECT_EQ(result.secondaries.data(),
+                  this->secondary_allocator().get().data()
+                      + result.secondaries.size() * i);
+
+        energy.push_back(result.secondaries[0].energy.value());
+        angle.push_back(celeritas::dot_product(
+            result.direction, result.secondaries.back().direction));
+    }
+
+    EXPECT_EQ(num_samples, this->secondary_allocator().get().size());
+
+    // Note: these are "gold" values based on the host RNG.
+
+    const double expected_energy[] = {
+        18872.0404050081, 43.592242238131, 4029.79130986756, 217.548454598548};
+
+    const double expected_angle[] = {0.999999971803589,
+                                     0.999999999587026,
+                                     0.999999999683752,
+                                     0.999999999474844};
+
+    EXPECT_VEC_SOFT_EQ(expected_energy, energy);
+    EXPECT_VEC_SOFT_EQ(expected_angle, angle);
+}
+
+TEST_F(RelativisticBremTest, stress_with_lpm)
+{
+    const int num_samples = 1000;
+
+    // Reserve  num_samples secondaries;
+    this->resize_secondaries(num_samples);
+
+    // Production cuts
+    auto material_view = this->material_track().material_view();
+    auto cutoffs       = this->cutoff_params()->get(MaterialId{0});
+
+    // Create the interactor
+    RelativisticBremInteractor interact(model_lpm_->host_ref(),
+                                        this->particle_track(),
+                                        this->direction(),
+                                        cutoffs,
+                                        this->secondary_allocator(),
+                                        material_view,
+                                        ElementComponentId{0});
+
+    RandomEngine& rng_engine = this->rng();
+
+    // Produce samples from the original incident angle/energy
+    real_type average_energy{0};
+    Real3     average_angle{0, 0, 0};
+
+    for (int i : celeritas::range(num_samples))
+    {
+        Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+
+        average_energy += result.secondaries[0].energy.value();
+        average_angle[0] += result.secondaries[0].direction[0];
+        average_angle[1] += result.secondaries[0].direction[1];
+        average_angle[2] += result.secondaries[0].direction[2];
+
+        EXPECT_EQ(result.secondaries.data(),
+                  this->secondary_allocator().get().data()
+                      + result.secondaries.size() * i);
+    }
+    EXPECT_EQ(num_samples, this->secondary_allocator().get().size());
```

Comment:
We ought to test the RNG rejection rate too, possibly at multiple incident energies? You can check the rate with
```
double(rng_engine.count()) / double(num_samples)
```

Reply:
Added an example test.